src/core/toga/__init__.py

```diff
@@ -100,4 +103,6 @@ def set_platform(module_name=None, local_vars=locals()):
             raise
 
 # On first import, do an autodetection of platform.
-set_platform()
+# Bypass this for docs.
+if not os.environ.get('IS_SPHINX'):
```

Comment:
The code is needed to load the appropriate platform backend. If you're on macOS, you need to have the toga-cocoa backend; if you're on Linux, you need toga-gtk, and so on. However, you only ever want to have to write `import toga` in user space, and preferably not have to "configure" the library before you import it.

There's already an environment variable for controlling which backend is loaded: `TOGA_PLATFORM `, which is evaluated [here](https://github.com/tonybaloney/toga/blob/e7f5f79fe2de1f39f73a5e71c9528d5f85c439dc/src/core/toga/__init__.py#L41). Rather than name sphinx explicitly, you could provide an `if platform == "dummy"` case that didn't load anything into the platform namespace, and put `os.environ["TOGA_PLATFORM"] = "dummy"` into the sphinx config.

Of course, if you've got another suggestion for how to handle this general problem, I'm all ears.